### DISCUSSION TITLE
Binding existing objects

### DISCUSSION THREAD

**Discussion by oschepkovdaniel** [gengyong/luaaa#25] [Q&A] [answered]
Hi! The library is very useful, but what about binding existing C++ objects (instances of classes) to lua? Is there any solutions for this?

**Comment by gengyong** [comment, marked as answer]
define a "ctor" for it, return the object instance. the "ctor" accept one or two functor which for alloc and free the object, if you set deleter functor as null, the object will never be GC from lua, this behavior fit for most singleton patern.

**Reply by zazahunter** [reply]
so you mean like make a new class and make it return a pointer to the instance??

**Comment by oschepkovdaniel** [comment]
Thanks for help =]

вт, 7 июн. 2022 г., 08:19 GENG Yong ***@***.***>:
